bin/setup.js

```diff
@@ -9,7 +9,19 @@ var isInstallIntoApp = /node_modules/.test(process.cwd())
 // This block only executes if Hoodie is installed with as a dependency
 if (isInstallIntoApp) {
   var pathToAppRoot = path.resolve('..', '..')
-  var packageJson = require(path.join(pathToAppRoot, 'package.json'))
+
+  // log warning if package.json cannot be found (hoodiehq/hoodie#751)
+  try {
+    var packageJson = require(path.join(pathToAppRoot, 'package.json'))
+  } catch (error) {
+    if (error.code !== 'MODULE_NOT_FOUND') {
+      throw error
+    }
+    log.warn('setup', 'Could not find package.json at ' + path.join(pathToAppRoot, 'package.json'))
+    log.warn('setup', 'You must manually set the start script in your app’s package.json to "hoodie" in order for "npm start" to work')
+
+    process.exit(0)
```

Comment:
I seem to be remembering a somewhat recent major version change in Node.js that `process.exit` might cause `console.log` messages to be swallowed because of some async operation. I think this note in https://nodejs.org/api/process.html#process_a_note_on_process_i_o refers to what I mean:

> Synchronous writes avoid problems such as output written with console.log() or console.error() being unexpectedly interleaved, or not written at all if process.exit() is called before an asynchronous write completes. See [process.exit()](https://nodejs.org/api/process.html#process_process_exit_code) for more information.

and from https://nodejs.org/api/process.html#process_process_exit_code

> It is important to note that calling process.exit() will force the process to exit as quickly as possible even if there are still asynchronous operations pending that have not yet completed fully, including I/O operations to process.stdout and process.stderr

Now the question here is if `log.warn()` is sync or async.

If they are, I think the solution is to register with the `beforeExit` event on `process` and do the desired output there: https://nodejs.org/api/process.html#process_event_beforeexit

* * *

And a matter of style, if `exit`ing because of an error condition, a return code other than `0` (zero) should be used. `1` would be an option.